ddl/set.lua

```diff
@@ -77,10 +79,22 @@ local function create_space(space_name, space_schema, opts)
     end
 
     if not is_dummy then
-        local ok, err = pcall(create_sharding_key, space_name, space_schema)
+        local ok, err = pcall(create_metadata, space_name, {space_schema.sharding_key}, "sharding_key")
         if not ok then
             return nil, string.format("spaces[%q].sharding_key: %s", space_name, err)
         end
+
+        local sharding_func = space_schema.sharding_func
+        local lua_code = false
```

Comment:
I would write a sharding function name into one field and sharding function body into another one. It seems logical to make both fields nullable. Mixing names and bodies into one field looks error prone for me.

Reply:
Updated:

`_ddl_sharding_func` space format now contains `sharding_func_name` and `sharding_func_body` nullable fields.
```lua
local function _set_schema(schema)
    ...
    set_metadata_space(
        'sharding_func',
        {
            {name = 'space_name', type = 'string', is_nullable = false},
            {name = 'sharding_func_name', type = 'string', is_nullable = true},
            {name = 'sharding_func_body', type = 'string', is_nullable = true},
        }
    )
    ...
end
```